Branch-SDK/src/main/java/io/branch/referral/Branch.java

```diff
@@ -1232,9 +1232,15 @@ public void logout() {
      * @param callback An instance of {@link io.branch.referral.Branch.LogoutStatusListener} to callback with the logout operation status.
      */
     public void logout(LogoutStatusListener callback) {
-        ServerRequest req = new ServerRequestLogout(context_, callback);
-        if (!req.constructError_ && !req.handleErrors(context_)) {
-            handleNewRequest(req);
+        prefHelper_.setInstallParams(PrefHelper.NO_STRING_VALUE);
+        prefHelper_.setSessionParams(PrefHelper.NO_STRING_VALUE);
+        prefHelper_.setIdentity(PrefHelper.NO_STRING_VALUE);
+        prefHelper_.clearUserValues();
+        //On Logout clear the link cache and all pending requests
+        linkCache_.clear();
```

Comment:
what does linkCache contain?

Reply:
It caches link returned by create url ( generateShortLink)

Reply:
This was done earlier like this :
<img width="702" alt="image" src="https://github.com/BranchMetrics/android-branch-deep-linking-attribution/assets/102190347/07725480-b2d7-42f0-8f22-6601362feb58">

Reply:
@JagadeeshKaricherla-branch  If installParams are cleared from the preferences, how we will get them again ? If user logs out once, API firstReferringParams will always return null.

Reply:
@NidhiDixit09  There is a check here for that use case 
https://github.com/BranchMetrics/android-branch-deep-linking-attribution/blob/c89fd4938276ee6b4a4a6cbef089688d40241ca7/Branch-SDK/src/main/java/io/branch/referral/Branch.java#L1453
Also, emptying install params was done earlier on API response, it got relocated now.
<img width="982" alt="image" src="https://github.com/BranchMetrics/android-branch-deep-linking-attribution/assets/102190347/86dab389-e8ae-43a2-9ed4-f36244122c4b">